config/secret_config.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/jahvon/tuikit/types"
+	"gopkg.in/yaml.v3"
+)
+
+type SecretConfig struct {
+	Name   string `json:"name"  yaml:"name"`
+	Secret string `json:"value" yaml:"value"`
+}
+
+type SecretConfigList []SecretConfig
+
+type enrichedSecretConfigList struct {
+	Secrets SecretConfigList `json:"secrets" yaml:"secrets"`
+}
+
+func (c *SecretConfig) AssignedName() string {
```

Comment:
This method is just returning the value of an exported struct field. It can be deleted and replaced with direct calls to that field. You may consider creating a constructor for the `Secret`/`SecretConfig` struct so that the name and value is ensured to be set once all secret creation uses it.

Reply:
You're right, these methods add unnecessary overhead. I'll remove them. I'll also add a constructor as you mentioned. (I'm assuming the constructor's role would be to validate the secret's reference, correct?)